src/Symfony/Component/DependencyInjection/Compiler/ResolveParameterPlaceHoldersPass.php

```diff
@@ -58,10 +58,13 @@ protected function processValue($value, $isRoot = false)
             return $this->bag->resolveValue($value);
         }
         if ($value instanceof Definition) {
+            // don't record these as new changes to the Definition
+            $value->setTrackChanges(false);
             $value->setClass($this->bag->resolveValue($value->getClass()));
             $value->setFile($this->bag->resolveValue($value->getFile()));
             $value->setProperties($this->bag->resolveValue($value->getProperties()));
             $value->setMethodCalls($this->bag->resolveValue($value->getMethodCalls()));
+            $value->setTrackChanges(true);
```

Comment:
This is a side-effect of the change tracking... changes. This wasn't actually needed to get the new behavior working correctly, it was just needed to get some tests to pass (i.e. the tests were compiling the container and then comparing Definition objects together... one of the Definition objects now had a few extra "changes")